xikolo-ios/ViewControllers/News/NewsViewController.swift

```diff
@@ -10,4 +10,65 @@ import UIKit
 
 class NewsViewController : AbstractTabContentViewController {
 
+    enum TabContent : Int {
+        case newsArticles = 0
+        case platformEvents = 1
+    }
+
+    @IBOutlet weak var containerView: UIView!
+
+    var containerContentViewController: UIViewController?
+    
+    @IBAction func switchViewControllers(sender: UISegmentedControl) {
+        if let position = TabContent(rawValue: sender.selectedSegmentIndex) {
+            updateContainerView(position)
+        }
+    }
+
+    override func viewDidLoad() {
+        updateContainerView(.newsArticles)
+        syncContent()
+    }
+
+    override func updateUIAfterLoginLogoutAction() {
+        super.updateUIAfterLoginLogoutAction()
+        syncContent()
+    }
+
+    func syncContent() {
+        NewsArticleHelper.syncNewsArticles()
+        PlatformEventHelper.syncPlatformEvents()
```

Comment:
In the case of a login or logout, we do. If the user visits the tab for the first time in this launch of the app, we want to already prepare the platform events so they are there when the user switches views. The reload of news articles is not necessary in this particular scenario. Do we care enough about this single request that it makes sense to change it?